server/selection_algorithm.go

```diff
@@ -58,7 +58,7 @@ func (sa ProbabilitySelectionAlgorithm) filterByPerfScore(ctx context.Context, a
 }
 
 func (sa ProbabilitySelectionAlgorithm) filterByMaxPrice(ctx context.Context, addrs []ethcommon.Address, maxPrice *big.Rat, prices map[ethcommon.Address]*big.Rat) []ethcommon.Address {
-	if maxPrice == nil || len(prices) == 0 {
+	if maxPrice.Num().Int64() == 0 || len(prices) == 0 {
```

Comment:
We should probably still have a nil check here for safety

Reply:
A more idiomatic way of doing this would also be
```suggestion
	if maxPrice == nil || maxPrice.Sign() == 0 || len(prices) == 0 {
```
(still need to check for `nil` tho yeah)

Reply:
Agree